tests/test_connector.py

```diff
@@ -200,3 +223,28 @@ def test_sync(self):
         mock_method.assert_called_once()
 
         self.connector_instance.publish_message = real_method
+
+    @pytest.mark.timeout(30)
+    def test_consumer_restarted(self):
+        self.connector_instance.register_consumer(name="test3", vhost=self.connector_instance.vhost,
+                                                  exchange='',
+                                                  queue='test_failing_once_queue',
+                                                  callback=self.connector_instance.callback_func_3,
+                                                  restart_attempts=1,
+                                                  auto_ack=False)
+        self.connector_instance.run_consumers(names=('test3',))
+        with self.connector_instance.create_mq_connection(vhost=self.connector_instance.vhost) as mq_conn:
+            self.connector_instance.emit_mq_message(mq_conn,
+                                                    queue='test_failing_once_queue',
+                                                    request_data={'data': 'knock'},
+                                                    exchange='',
+                                                    expiration=4000)
+            self.connector_instance.consumer_restarted_event.wait(self.connector_instance.observe_period + 5)
+            time.sleep(3)
```

Comment:
is this extra sleep necessary? Looks like we are already waiting for the listener to restart

Reply:
> is this extra sleep necessary? Looks like we are already waiting for the listener to restart

we need some tolerance period to await for consumer restart (establishing mq connection), otherwise that would break

Reply:
I see, so if a message is emitted while the consumer is restarting, is that message not handled?